components/Header.js

```diff
@@ -0,0 +1,73 @@
+import NavBarStyles from "./styles/NavBarStyles";
+import SignUpContainerStyles from "./styles/SignUpContainerStyles";
+import Link from "next/link";
+import Cart from "./Cart";
+import { useDispatch, useSelector } from "react-redux";
+import { setCartOpen, signOut } from "../actions";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import HeaderStyle from "./styles/HeaderStyle";
+import Logo from "./styles/LogoStyle";
+
+export default function Header() {
+  const dispatch = useDispatch();
+  const router = useRouter();
+  const cartOpen = useSelector((state) => state.categories.cartOpen);
+  const cartItems = useSelector((state) => state.categories.cartItems?.cart);
+  const userId1 = useSelector((state) => state.categories.userId);
+  const [userId, setUserId] = useState(userId1);
+  useEffect(() => {
+    setUserId(localStorage.getItem("user"));
+  }, [userId]);
+  const SignOut = () => {
+    dispatch(signOut());
```

Comment:
dispatch should not be used here. As discussed use mapDispatchToProps to dispatch actions from components.

Reply:
done